### PR TITLE
Text processing fixes

### PR DESCRIPTION
**Pull request by Gillibald**
## What does the pull request do?

Fix RightToLeft selection
Fix dynamic inlines
Fix font fallback
Prevent a crash when no TextTrimming is set

## Checklist

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022712-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022714-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022742-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

File: src/Avalonia.Base/Media/TextFormatting/TextLineImpl.cs [hunk 1]
```diff
@@ -211,12 +260,16 @@ private static CharacterHit GetRunCharacterHit(DrawableTextRun run, int currentP
...
-                        if (!shapedRun.GlyphRun.IsLeftToRight)
+                        if (shapedRun.GlyphRun.IsLeftToRight)
                         {
-                            offset = Math.Max(0, offset - shapedRun.Text.End);
+                            offset = Math.Max(0, currentPosition - shapedRun.Text.Start);
                         }
+                        //else
+                        //{
+                        //    offset = Math.Max(0, currentPosition - shapedRun.Text.Start + shapedRun.Text.Length);
+                        //}
```
Comment (maxkatz6) [inline src/Avalonia.Base/Media/TextFormatting/TextLineImpl.cs:272]: Should it be commented?

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0022753-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Base/Media/TextFormatting/TextLineImpl.cs (hunk 1 above)
Comment (Gillibald) [reply src/Avalonia.Base/Media/TextFormatting/TextLineImpl.cs:272]: Will remove it

Comment (Gillibald) [reply src/Avalonia.Base/Media/TextFormatting/TextLineImpl.cs:272]: #8671

**Pull request merged**